lib/dao.js

```diff
@@ -840,6 +840,19 @@ DataAccessObject.replaceOrCreate = function replaceOrCreate(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId && id) {
+    this.findById(id, function(err, instance) {
```

Comment:
removing the id check used to break the previous tests when forcedId was set to false, however it does not need that anymore, so I don't think I need it for now. I will keep an eye on it.